packages/jest-jasmine2/src/jasmineAsyncInstall.ts

```diff
@@ -59,7 +64,7 @@ function promisifyLifeCycleFunction(
     // didn't return a promise.
     const asyncJestLifecycle = function (done: DoneFn) {
```

Comment:
This (and it's counterpart in `promisifyIt`) are whats making TS sad, and where it looks like there could be an incorrect type somewhere.

In `Env.ts`, the hooks & spec functions (i.e `afterAll`, `beforeEach`, `it`, `fit`, etc) are typed to take a function parameter of type `QueueableFn['fn']`, but here `DoneFn` is used, which is slightly different.

Each section of code heavily uses the differences between the two types (i.e here, `done.fail` is used, and in `Env` the types eventually end up at the `queueableFn` property on the `Attributes` interface).

Looking at `@types/jasmine`, it looks like the `DoneFn` might be the more correct type in most cases, but this is where I think I'm missing an important puzzle piece that'd make everything click :)

Reply:
I know very little about jasmine, unfortunately... @doniyor2109 you migrated this module back in #7970 - do you remember/have any context?

Reply:
@SimenB @G-Rath I am not sure either. I have created a separate `DoneFn` type because `QueueableFn['fn']` did not include `fail` method.

Reply:
I think I've found it, and it's actually simple:

```
interface DoneFn {
  (error?: any): void;
  fail: (error: Error) => void;
}

export type QueueableFn = {
  fn: (done: DoneFn) => void;
  timeout?: () => number;
  initError?: Error;
};
```

I noticed when doing #10215 that `queueRunner` does `next.fail = function ...` and then does `fn.call(options.userContext, next)`, so `QueueableFn['fn']` should just take `DoneFn`, which in turn should just support `error?: any` as a parameter.

Doing that, and replacing the right spots with `DoneFn` makes it all come together - the only problem is that then passing the noops is a problem because they require `.fail` 😬  (but that can be fixed using a variable)